Sources/Datadog/DatadogInternal/DatadogCoreProtocol.swift

```diff
@@ -170,8 +170,12 @@ public protocol FeatureScope {
     ///   - bypassConsent: `true` to bypass the current core consent and write events as authorized.
     ///                    Default is `false`, setting `true` must still respect user's consent for
     ///                    collecting information.
+    ///   - forceNewBatch: `true` to enforce that event will be written to a separate batch than previous events.
+    ///                     Default is `false`, which means the core uses its own heuristic to split events between
+    ///                     batches. This parameter can be leveraged in Features which require a clear separation
+    ///                     of group of events for preparing their upload (a single upload is always constructed from a single batch).
     ///   - block: The block to execute.
-    func eventWriteContext(bypassConsent: Bool, _ block: @escaping (DatadogContext, Writer) throws -> Void)
+    func eventWriteContext(bypassConsent: Bool, forceNewBatch: Bool, _ block: @escaping (DatadogContext, Writer) throws -> Void)
```

Comment:
Good idea 👍. Turns out that methods in protocol extensions can define default parameter values, so I was able to avoid introducing new methods. Done ✅.